src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -18,17 +19,36 @@ public struct StringValues : IList<string>, IReadOnlyList<string>, IEquatable<St
 
         private readonly string _value;
         private readonly string[] _values;
+        private readonly Encoding _encoding;
+        private readonly byte[] _bytes;
+
+        public StringValues(string value, Encoding encoding)
+        {
+            _value = value;
+            _values = null;
+            _encoding = encoding;
+            _bytes = _encoding.GetBytes(value);
+        }
 
         public StringValues(string value)
         {
             _value = value;
             _values = null;
+            _encoding = null;
+            _bytes = null;
         }
 
         public StringValues(string[] values)
         {
             _value = null;
             _values = values;
+            _encoding = null;
+            _bytes = null;
+        }
+
+        public static StringValues CreateConstant(string value, Encoding encoding)
```

Comment:
I think "CreateXxx" and "TryGetXxx" should have the same "Xxx" in the name. The "Xxx" doesn't have to be "Constant". 

That term was chosen originally because of how it was used: `internal static class Constants { public static readonly StringValues TextPlain = StringValues.CreateConstant("text/plain"); }`

Actually, wasn't it originally CreateBinaryConstant? How about `CreateBinary` and `TryGetBinary`? Or like you said `CreateBytes` and `TryGetBytes`?

The problem is: we *don't* want people to use the Create method *unless* they are creating a static readonly instance used the same way you use a constant.

If you have someone call StringValues.Create(foo) per-request - as if it was just an alias for the ctor - they'll actually be increasing their allocations per-request. That's why the "Constant" hint was in there: this method is for creating constants...